plugins/outputs/azure_monitor_logs/azure_monitor_logs_test.go

```diff
@@ -0,0 +1,277 @@
+			},
+		},
+		{
+			name:       "429 status is an error",
+			statusCode: http.StatusTooManyRequests,
+			errFunc: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+		{
+			name:       "500 status is an error",
+			statusCode: http.StatusInternalServerError,
+			errFunc: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+		{
+			name:       "503 status is an error",
+			statusCode: http.StatusServiceUnavailable,
+			errFunc: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+	}
+
+	plugin := &AzLogAnalytics{
+		CustomerID: "dummy_id",
+		SharedKey:  "ZHVtbXlfa2V5",
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(tt.statusCode)
+			})
+
+			err = plugin.Connect()
+			require.NoError(t, err)
+
+			// override write url
+			plugin.serviceURL = url.String()
+
+			err = plugin.Write([]telegraf.Metric{getMetric("cpu")})
+			tt.errFunc(t, err)
+		})
+	}
+}
+
+func TestWrite(t *testing.T) {
+	readBody := func(r *http.Request) ([]map[string]interface{}, error) {
+		buffer, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		var amm []map[string]interface{}
+		err = json.Unmarshal(buffer, &amm)
+		if err != nil {
+			return nil, err
+		}
+
+		return amm, nil
+	}
+
+	ts := httptest.NewServer(http.NotFoundHandler())
+	defer ts.Close()
+
+	url, err := url.Parse(fmt.Sprintf("http://%s", ts.Listener.Addr().String()))
+	require.NoError(t, err)
+
+	tests := []struct {
+		name    string
+		metrics []telegraf.Metric
+		handler func(t *testing.T, w http.ResponseWriter, r *http.Request)
+	}{
+		{
+			name: "metric with different names sent in different requests",
+			metrics: []telegraf.Metric{
+				getMetric("cpu"),
+				getMetric("mem"),
+			},
+			handler: func(t *testing.T, w http.ResponseWriter, r *http.Request) {
+				azmetrics, err := readBody(r)
+				require.NoError(t, err)
+				require.Len(t, azmetrics, 1)
+				w.WriteHeader(http.StatusOK)
+			},
+		},
+		{
+			name: "single azure metric",
+			metrics: []telegraf.Metric{
+				getMetric("cpu_test"),
+			},
+			handler: func(t *testing.T, w http.ResponseWriter, r *http.Request) {
+				// Header should be camel case
+				require.Equal(t, "CpuTest", r.Header.Get("Log-Type"))
+
+				// body should only contain 1 metric
+				azmetrics, err := readBody(r)
+				require.NoError(t, err)
+				require.Len(t, azmetrics, 1)
+
+				// check host value
+				value, ok := azmetrics[0]["Computer"]
+				require.True(t, ok) // key exists
+				require.Equal(t, "test-machine", value)
+
+				// check env value
+				value, ok = azmetrics[0]["Env"]
+				require.True(t, ok) // key exists
+				require.Equal(t, "testing", value)
+
+				// check metric value
+				value, ok = azmetrics[0]["Value"]
+				require.True(t, ok) // key exists
+				require.Equal(t, 42.0, value)
+
+				w.WriteHeader(http.StatusOK)
+			},
```

Comment:
Please write one handler function and directly specify. Furthermore, you might want to test the received data outside of the http-test-server to simplify code.